src/Aspire.Hosting.Azure.ContainerApps/AzureContanierAppsInfrastructure.cs

```diff
@@ -698,9 +712,36 @@ BicepValue<string> GetHostValue(string? prefix = null, string? suffix = null)
                 throw new NotSupportedException("Unsupported value type " + value.GetType());
             }
 
-            private BicepParameter AllocateVolumeStorageAccount(ContainerMountType type, string volumeIndex)
+            private BicepParameter AllocateVolumeStorageAccount(ContainerMountType type, string volumeIndex) =>
+                AllocateParameter(VolumeStorageExpression.GetVolumeStorage(resource, type, volumeIndex));
+
+            private BicepValue<string> AllocateKeyVaultSecretUriReference(BicepSecretOutputReference secretOutputReference)
             {
-                return AllocateParameter(VolumeStorageExpression.GetVolumeStorage(resource, type, volumeIndex));
+                if (!KeyVaultRefs.TryGetValue(secretOutputReference.Resource.Name, out var kv))
+                {
+                    // We resolve the keyvault that represents the storage for secret outputs
+                    var parameter = AllocateParameter(SecretOutputExpression.GetSecretOuputKeyVault(secretOutputReference.Resource));
+                    kv = KeyVaultService.FromExisting($"{parameter.ResourceName}_kv");
+                    kv.Name = parameter;
+
+                    KeyVaultRefs[secretOutputReference.Resource.Name] = kv;
+                }
+
+                if (!KeyVaultSecretRefs.TryGetValue(secretOutputReference.ValueExpression, out var secret))
+                {
+                    // Now we resolve the secret
+                    secret = KeyVaultSecret.FromExisting($"{kv.ResourceName}_{secretOutputReference.Name}");
+                    secret.Name = secretOutputReference.Name;
+                    secret.Parent = kv;
+
+                    KeyVaultSecretRefs[secretOutputReference.ValueExpression] = secret;
+                }
+
+                // REVIEW: Is there a better way to do this?
+                return new MemberExpression(
```

Comment:
cc @tg-msft @eerhardt

Reply:
Since this property is nested under "Properties", there isn't a better way to do this right now. But @tg-msft will fix it so this can be `return secret.SecretUri;`.

Instead of `REVIEW` can you change this to `TODO`? That's how I've been tracking all the things that should be fixed when the CDK 1.0 is out.